### ISSUE TITLE
You can't disable notifications for pinned messages in group conversations

### ISSUE THREAD

**Issue by maxklock** [telegramdesktop/tdesktop#1864]
### Steps to reproduce
1. Open an Group conversation
2. Disable notifications
3. Get an new pinned message
### Expected behaviour

Tell us what should happen

No notification
### Actual behaviour

Tell us what happens instead

Notification
### Configuration

**Operating system:**
Windows10
**Version of Telegram Desktop:**
0.9.40
### Logs

Insert logs here (if necessary)

You can type "debugmode" in settings and then see ~/.TelegramDesktop/DebugLogs/log_...txt for log files.
Type "debugmode" in settings again to disable logs.

**Comment by john-preston** [comment]
@maxklock Notifications about pinned messages are treated like notifications from messages that mention you explicitly or reply to you explicitly. In this case, if the notifications for the group are disabled, the notifications settings for the admin are used. You can disable notifications from the admin who pinned the message if you don't want to be notified from him.

**Comment by maxklock** [comment]
Why? When notifications in a group are disabled I don't want to get any notification. And disable the other person isn't a way to do that, because I want to get normal notifications from the person, when not in that group

**Comment by stek29** [comment]
@auchri [api issue] ?

**Comment by john-preston** [comment]
@maxklock In Telegram there is a way to notify user in a group even if he muted that group, but not muted you — by mentioning a user in a message or by replying to him. Then he will be notified, if he muted the group, but didn't mute you.

Exactly the same logic goes with the pinned messages (anyway, it is mostly a server-side logic, not about Telegram Desktop) — when admin is pinning a message, he can choose: notify all members or not. If he notifies all members, then they all are notified, if they didn't mute him. If he decides just to pin the message, only the guys who didn't mute the group, are notified.

**Comment by VBelozyorov** [comment]
Guys, for `v0.9.42` it seems that `Notify all members` checkbox is simply inverted :)

I'm talking about this one:
![notify](https://cloud.githubusercontent.com/assets/860208/14560708/5e8a4b5a-0322-11e6-9900-d45142f72e5d.png)

If I set it to `unchecked`, other supergroup members say they recieved notification, but when I leave it `checked` — they hadn't recieved anything. 
This behaviour is opposite official Android and iOS clients.
Looks like it simple typo, isn't it?

**Comment by strangeattr** [comment]
It drives me crazy every time when I am busy with something important (not sufficiently to turn off the phone completely though), the phone starts signalizing about a new message, I hurry to take a look what's there and that's somebody having pinned "Yo-ho-ho" (or something of equal importance) in a funny pictures channel I have joined just to look at it a couple of times a month when I happen to be both tired and bored. Approximately 98 of 100 pinned messages are plain nonsense and not even fun actually (however I still believe the feature can be used for good and should not be eliminated even though it is misused so widely).

It also seems highly illogical that I can mute a person completely (so I will only see their messages when I intentionally go into it) but can't really mute a chat.

An option to mute a chat completely (including pinned message mute) is a feature I miss CRAZILY. I think it is CRUCIAL and lack of it is an extreme, almost intolerable annoyance. I feel like I will have to leave all chats unless this feature gets added or I find a workaround.

However, this only applies to the mobile Telegram client actually, but they haven't got a public issue tracker and this issue seems very relevant so I have decided to express my opinion here. 

PS: It also looks ridiculous when a pinned message arrives from a muted chat and a badge appears over the Telegram icon saying there are thousands of unread messages.

**Comment by Squirrel2020** [comment]
Telegram developers are idiots, especially Pavel Durov because he is letting such little thing stand in a way of user adoption.

>>Do I really have to leave a channel to avoid this when my only intention was to read it on occasion when I have mood?

Totally agree.

I am quitting telegram forever.

**Comment by stek29** [comment]
@Squirrel2020 https://telegram.org/deactivate btw

**Comment by Stauc** [comment]
please fix this, it's annoying !!!

**Comment by stek29** [comment]
Can be in the same menu as #3786

**Comment by rivf** [comment]
Sadly this feature is used for spamming with ads in some interesting supergroups. Very annoying.

**Comment by rvaleyev** [comment]
Hi, how can i Turn on back notifications by pin? Its very important for me

**Comment by Su4oNoK** [comment]
Hi, how can i Turn on back notifications by pin if notification is disabled on group chat? Its very important to me
I have same problem

**Comment by alic3woodstock** [comment]
"Discord" is a great example you can disable "@everyone" and "@here" (equivalent of pinned messages) and still get the notification when you are mentioned.

**Comment by miconian** [comment]
This is a ridiculous deal-breaker. When the issue is fixed, I will consider re-installing the app. 

Last night, there was the possibility of an important alert from a specific telegram group, so I slept near my phone, with the sound on, and notifications for the app on. But I kept getting woken up by the dings of pinned posts in another group. I tried to leave that other group, but using my phone it was too difficult, as there are tens of thousands of members, and it took too long to scroll past them all to the "delete" option that is ludicrously located at the bottom of the member list. I was one of several members of that group asking the admin to stop with the mandatory notifications, but he/she was unaware of this stupid bug.

The fact that this has been open for so long is mind-boggling. This is 1980s-level user interface incompetence.

**Comment by auchri** [comment]
@miconian Well, you can just use the three dot menu at the top right at the Android app to leave the group. So you don't have to scroll through all the members.

Also, this is not a bug.

**Comment by miconian** [comment]
I don't have Android, I have an iPhone. The three dot menu is not available.

I eventually removed myself from the group in question using the three-dot menu at the top of the Mac desktop app. But that doesn't change the fact that the issue could arise again on some other channel. 

I'm sorry for calling it a stupid bug. I should have said "stupid feature."

**Comment by RememberTheAir** [comment]
I agree with those who say that not being able to mute pinned messages doesn't make sense. Having the "notify with sound" option always checked by default makes sense considering the [purpose of the feature](https://telegram.org/blog/supergroups5k), but having it unchecked instead would save me from a lot of unwanted pop-ups.

Challegram has an option to disable pinned message notifications in the chat settings, and notifications there are much much less toxic than on other clients.

Telegram Desktop can't do nothing about this, because of concistency of the clients ecosystem, beside adding an option to mute them all as the Android app does. I just hope one day the API will support pinned messages notifcations  settings :(

**Comment by ttygde** [comment]
very annoying, please fix

**Comment by ZitRos** [comment]
I would appreciate Telegram adding more options about what to mute. Pinned posts in particular — they are very annoying when you have 20+ channels where you occasionally read news or sleep until @mentions. Also consider the behavior when such muted chats does not bubble up to the top in the menu. I always need to scroll down 2 screens to finally reach a conversation with my friend which happened just a couple of tens of minutes ago.

**Comment by macelii** [comment]
It's now 2018, this insane logic doesn't seem to get a chance to be fixed though.
I'll just leave telegram world for now then.

**Comment by brunogarciacosta** [comment]
You can disable it on the phone's app, but not on the desktop version.

¯\_(ツ)_/¯

**Comment by davisonio** [comment]
> You can disable it on the phone app, but not on the desktop version.
> 
> ¯_(ツ)_/¯

How? I don't see it

**Comment by brunogarciacosta** [comment]
> How? I don't see it

Configs > Notification and Sound > Events > Pinned Messages

**Comment by ZitRos** [comment]
> Configs > Notification and Sound > Events > Pinned Messages

Unfortunately, it's not on iOS:

![image](https://user-images.githubusercontent.com/4989256/36495616-81ab101a-173e-11e8-97c4-baf9423e26f8.png)

**Comment by davisonio** [comment]
That's awful yeah I'm using iOS

**Comment by blaaaaaaa** [comment]
+1 very annoying

**Comment by moltar** [comment]
Getting spammed to death by channels with pinned messages. They definitely have caught onto this and using it as a spamming tool.

**Comment by john-preston** [comment]
@moltar Pinned messages in channels that are muted are not delivering any notifications, at least on tdesktop. Maybe you meant groups not channels?

**Comment by moltar** [comment]
Right. This actually happens on iOS. I was just Googling the problem and stumbled on this issue. Figured maybe they share functionality and if I complain here, iOS will get fixed too. Sorry for the noise.

**Comment by utdrmac** [comment]
Please fix this for iOS. Extremely annoying to be continuously bombarded with notifications (usually pins) from groups that I've muted. What's the point of even having the mute option if it won't actually mute everything about that group?

**Comment by davisonio** [comment]
this is a joke

**Comment by dragGH102** [comment]
Same on Android.

**Comment by utdrmac** [comment]
@john-preston Can you please comment? This is an extremely old issue that is still not resolved.

**Comment by john-preston** [comment]
@utdrmac This works consistent with another official apps, any admin that pins message has an ability to mark it "notifiable", which works the same way as if admin has mentioned you in this message — you receive a notification if you didn't disable notifications from this chat and from this admin (both). So this is a special way to mention all your users at once.

**Comment by brunogarciacosta** [comment]
@john-preston The official Android app has the feature to disable pinned messages notifications on groups.  I don't understand the reasoning behind not adding this simple feature. 
Sometimes there are multiple admins on a group, am I supposed to mute each admin everytime I join a group? (Not to mention admins that have been added recently and I didn't have the chance to mute.)
What if I know that admin and I don't want to mute his private messages?

**Comment by utdrmac** [comment]
@john-preston Why must I disable from TWO different locations? I don't want that. That is incredibly unintuitive. "Disable Notifications" has a very simple meaning: It means "turn off all notifications". It does not mean "turn off _some_ notifications". Like a light switch, either on or off, not dimmed 1/2 way.

I don't care what the group admins set/do on their side. On my side, on **MY DEVICE**, on **my settings**, I have it set to disabled so the app should honor **me** and what **I want** and not the other guy.

**Comment by john-preston** [comment]
@utdrmac Well, the same logic can be applied to mentions. So if someone mentions you on Telegram chat and you receive notifications you may come here as well and write the same text (about disabling two different settings, turn off all notifications etc). But the feature is considered usefull by the team. I guess the same goes with those pinned messages notifications.

I've heard your arguments and I'll ask once more about that.

**Comment by galloclaudio** [comment]
Any know other telegram desktop? FUCK the official if that not include a SIMPLE swtich off button to disable the pinned message notify. Its a mindfuck think the team dont do that button. Why? why u hate ur users?

**Comment by zeroone2numeral2** [comment]
I share all the angry feelings about pinned messages notifications. I hate them, with all my heart. But people have to understand that it's up to Telegram (as a messagging platform) to introduce a notification setting (global/per-group) to toggle them on or off. TDesktop needs to stick to what the API offers. No API method to turn notifications off -> no option to do that on TDesktop. The Android app has a setting to turn them off, for some reasons, but that setting is not synced across clients and it shouldn't be there, since if I turn off a notification setting on a cloud-based app, I expect to not receive notifications regardless of the client I am using. It is completely ok for TDesktop to not offer a client-side option to turn pinned message notifications off.

That said, I really hope Telegram will do something about these notifications. I just don't understand why they thought that sending a notification to up to 100.000 people at time was a good idea. 99% of the times these notifications are just unwanted phone rings. I can understand there could be something _so_ important that an admin may want to notify every member of a groupchat in a single shot. But at least give us an option to turn these notifications off without having to go through the notification settings of all the admins. Really, what would be the downside of a per-chat, on-by-default pinned message setting?

**Comment by Romeno** [comment]
Why not to fix this?

**Comment by zeroone2numeral2** [comment]
@Romeno there is nothing to fix on TDesktop's side. And apparently Telegram thinks there is nothing to fix in general. They designed pinned messages as something that should _always_ notify group members, and they don't think this is something that needs changes, despite it overrides users notification settings of their groupchats. Sadly :(

**Comment by galloclaudio** [comment]
we can press for this to be arranged in the telegram client just as they have done in the mobile phone. As much as it depends on telegram you can make an option from the application to REALLY silence the whole group and surely there are few lines of code

**Comment by Romeno** [comment]
@zeroone2numeral2 sure, but apparently users think different. This just does not make sense? Why users need to suffer or limit their group inclusion? Why should one leave the group if he just does not want the notifications? Do you think it is sane?

**Comment by zeroone2numeral2** [comment]
@Romeno I absolutely agree with you, I don't like forced notifications and I think most of us don't too. But this is out of the scope of this client. This is something that should be implemented on the backend first

@galloclaudio then we should press them on Twitter. Local settings are not synced, and these kind of things (notifications settings) on a cloud-based service should be synced across all the clients. I use Telegram from a number of platforms and having to toggle/untoggle the same setting from all of them because it is not synced sounds like a poor implementation of a feature that should be included in the API. At the moment it is not. As I said, the android client, as a client of a cloud-based service, shouldn't implement this kind of solutions if they are not synced across devices, because that's what Telegram users would expect: disable notifications once from a client - they are disabled in all the other apps too. Apparently DrKlo doesn't like pinned messages notifications too :)

Anyway, I wouldn't dislike a local setting for pinned messages notification on TDesktop. The fact is that having that setting availble on the two most used Telegram clients (android, desktop) whithout it being synced wouldn't just make sense - why not just implementing it at protocol level? That's why I think making pressure on TDesktop about this doesn't make much sense and just brings more fragmentation among clients/annoyance for users who have to disable that setting from all the Telegram apps they use

**Comment by Romeno** [comment]
@zeroone2numeral2 pushing this issue on twitter is a good idea, though I dont think it is out of scope for the TDesktop client cause this feature is implemented for Android app. I can disable pinned messages there and actually I did it long time ago. But on desktop I keep getting the notifications from the groups I dont need. I dont see any justification for this... Why it couldn't be fixed on desktop as well. 

Syncing notification settings and protocol level thingy you are talking about is a good idea though.

**Comment by Keltere** [comment]
Please implement it also on TDesktop thanks

**Comment by youurayy** [comment]
Still getting bothered by this to hell on the desktop and disabling notifications on admin is just plain retarded. FIX THIS.

**Comment by rromm** [comment]
> Notifications about pinned messages are treated like notifications from messages that mention you explicitly or reply to you explicitly

and it's stupid — I'm not interested in group's pins and admin wouldn't FORCE (unlike TG does) message to not interested ppl.

**Comment by RememberTheAir** [comment]
> He should do anything to fix this, even if boss told him to not fix

We're not talking about some nice third-party client, we're talking about the second (I believe?) most used **official** client of a messaging service used by 200 millions people. You can't just change how a feature works on a single client and sacrifice consistency just because there's an issue on that client's repo with barely 50 likes. This thing needs to be fixed _everywhere_ (yes I hate these unwanted notifications too), not just on desktop. TDesktop can't just ignore how "Telegram" decided how this feature is supposed to work. Just imagine if every client started to implement every possible request filed by the community on its own, without respecting any kind of consistency with the tg API. I want my settings to be synced and I want to find them on every Telegram client I use - or at least on the official ones. That's a core principle of a multiplatform service. I'm kind of surprised no one made a fork of TDesktop where you can disable pinned messages notifications yet

**Comment by rromm** [comment]
[RememberTheAir](url), it's wrong quotation:

> He should do anything to fix this, even if boss told him to not fix

right is:

> even if boss told him to not fix — he could spam boss by ...

> You can't just change how a feature works 

but you can add future to muthen such notifies.
tl;dr.

**Comment by Mirarkitty** [comment]
I want to be able to mute ALL notifications in a channel, popus as well as @mirar. I can understand the rationale behind it not being there, but that doesn't mean it's not a bother that it's missing the possibility to muting notifications.

**Comment by Keltere** [comment]
> @Mirarkitty you can already mute all notifications coming from channels. Pinned messages in channels won't produce any notification if you muted the channel as a whole

Just on smartphone, on telegram desktop you can't.

**Comment by RememberTheAir** [comment]
@Keltere yeah you can, you don't receive pinned messages notifications in channels, if you muted it. I just relized @Mirarkitty might not be talking about _channels_, but groups. Channels and group chats are a different thing on Telegram. You always receive pinned messages notifications from groups (unless you are using Android when the message is pinned and you turned them off), but with channels they always follow the channel's notification settings

**Comment by Keltere** [comment]
> @Keltere yeah you can, you don't receive pinned messages notifications in channels, if you muted it. I just relized @Mirarkitty might not be talking about _channels_, but groups. Channels and group chats are a different thing on Telegram. You always receive pinned messages notifications from groups (unless you are using Android when the message is pinned and you turned them off), but with channels they always follow the channel's notification settings

yes i think me and him meant to say groups, sorry

**Comment by iJaack** [comment]
> @maxklock In Telegram there is a way to notify user in a group even if he muted that group, but not muted you — by mentioning a user in a message or by replying to him. Then he will be notified, if he muted the group, but didn't mute you.
> 
> Exactly the same logic goes with the pinned messages (anyway, it is mostly a server-side logic, not about Telegram Desktop) — when admin is pinning a message, he can choose: notify all members or not. If he notifies all members, then they all are notified, if they didn't mute him. If he decides just to pin the message, only the guys who didn't mute the group, are notified.

That's the way I solved it, thanks! but I think there need to be a further setting to mute/unmute admin push notifications.

**Comment by VSG24** [comment]
Stupid, just fucking stupid...

**Comment by VSG24** [comment]
So stupid of Telegram...

**Comment by mervick** [comment]
I want to stop get notifications from groups without to leave them. 
What is the problem to add this setting?

**Comment by romrozen** [comment]
It's a bug - the users said.
It's a feature! - the developers exclaimed.

Seriously, quite odd that the stance of the devs remains after such a significant firm say from its users.

**Comment by neekless** [comment]
Official telegram android has it so I don't understand why tdesktop doesn't. The argument of consistency with other clients fail here.
I only want notifications that target me specifically - mentions and replies. Pinned messages are not targeting me specifically. The argument of using mentions logic also fail here.
Crazy how this annoying issue has persisted for 3 years yet the dev still thinks this is the correct behavior.

**Comment by brunogarciavaz** [comment]
> Official telegram android has option to disable pinned message notifications so I don't understand why tdesktop doesn't. The argument of consistency with other clients fail here.
> If I turned off group notifications, I only want notifications from a group that target me specifically - mentions and replies. Pinned messages are not targeting me specifically. The argument of using mentions logic also fail here.
> Crazy how this annoying issue has persisted for 3 years yet the dev still thinks this is the correct behavior.

It's the official telegram API that doesn't allow it, so I think this issue has little purpose (as tdesktop devs can't do anything about it). Does anyone know any feedback email we could use to contact the official telegram dev team?

**Comment by NexonSU** [comment]
> maxklock opened this issue on 6 Apr 2016
LOL. This client discontinued or what?

**Comment by SigmaNetwork** [comment]
Developers still being idiots I see and not implementing this

**Comment by antidotcb** [comment]
Any progress on that?

**Comment by RememberTheAir** [comment]
@antidotcb there isn't going to be any progress on this 8especially on tdesktop's side). This is how the thing is designed to work. It has been 2 years since pinned messages has been introduced, it's clear Telegram doesn't want to take a step back and give us an option to disable these notifications

**Comment by Remu-rin** [comment]
> it's clear Telegram doesn't want to take a step back and give us an option to disable these notifications

Then why official android client does have this option?

**Comment by RememberTheAir** [comment]
@Remu-rin the Android app has an option to mute pinned message notifications for some reasons. My guess is that the Android dev thought these notifications were annoying too and added it. Or because Telegram wanted to give most of the userbase the ability to hide them completely, thinking that the "all or nothing" deal would have pushed most of the people who were aware of that option to keep them enabled.
Anyway, the fact that neither any of the official apps (but the android ones) nor the API allow to configure this option shows that Telegram wants them to work this way. After all that's how they advertised them in their blogpost when the feature were introduced, that is, a way for admins to be sure to reach all the group members. We can debate whether this is legit or not (imo it isn't), but that's how they wanted this to be because, to date, the option on the android app hasn't been implemented platform-wide

**Comment by RememberTheAir** [comment]
Whoops, good news. Is it synced?

**Comment by john-preston** [comment]
@RememberTheAir No.

**Comment by NightMachinary** [comment]
I fail to understand this argument that since Telegram for some reason wants the clients to work some way, the clients should not introduce options to work in other ways. As long as the other way is not abusing other users (e.g., ghost mode), Telegram is just being like a third-world immature leader by enforcing its desires upon the users. A significant amount of people have demonstrated that they hate this design decision, and they bloody don’t want enforced spam on their personal phones (THEIR phone not anyone else’s).
The apologetics on this issue are just mind-boggling. I recommend you watch this video, it gives one a sense of the sheer insanity of current paternalistic (and exploitative, occasionally) worldviews entrenched in today’s tech scene. 
https://www.youtube.com/watch?v=fkkDvKGcNSo

PS: I’m sure if most official clients implemented this, then the API would follow. And as the passage of years have shown, bugging Telegram on Twitter does not work.

**Comment by alexb5dh** [comment]
@NightMachinary, this seems to be already implemented in desktop client and will probably be released soon.

**Comment by Keltere** [comment]
@alexb5dh yes was added in this commit https://github.com/telegramdesktop/tdesktop/commit/af85aec33b86f138346f28ba70303a0e590f3db7 by john.
I'm testing the beta 1.7.1 and the "disable pinned message" seem to work just fine.
At last our complains were heard.

**Comment by shideneyu** [comment]
So 4 years after, we really can't disable pin message notifications ? I really need to leave the groups that spams the pinned message ? Bullshit

**Comment by tandav** [comment]
is this option available in macOS telegram?

**Comment by RememberTheAir** [comment]
> is this option available in macOS telegram?

If you're using "Telegram Light" from the store (which is tdesktop), then yes. You should ask in [@TelegramMacOSTalk](https://t.me/TelegramMacOSTalk) or on the MacOS's app GitHub repository
